api/src/app/modules/public-comment/public-comment.service.ts

```diff
@@ -149,11 +209,93 @@ export class PublicCommentService extends DataService<PublicComment, Repository<
     }
 
     const recordIds = _.map(records, 'id');
-    const decryptedColumnCollection = await this.obtainDecryptedColumns(recordIds);
+    const decryptedColumnCollection = await this.obtainDecryptedColumns(
+      recordIds
+    );
     return records.map((r) => {
-      const decryptedPartial = _.find(decryptedColumnCollection, {'id': r.id});
+      const decryptedPartial = _.find(decryptedColumnCollection, { id: r.id });
       Object.assign(r, decryptedPartial);
       return this.convertEntity(r);
     });
   }
+
+  /**
+   * Returns the total number of public comments grouped by district.
+   * Used by analytics dashboard module.
+   *
+   * @param startDate - Start of date range (YYYY-MM-DD)
+   * @param endDate - End of date range (YYYY-MM-DD)
+   * @returns Promise resolving to an array of PublicCommentCountByDistrictResponse
+   */
+  async getCommentCountByDistrict(
+    startDate: string,
+    endDate: string
+  ): Promise<PublicCommentCountByDistrictResponse[]> {
+    return await this.repository
+      .createQueryBuilder('c')
+      .where('c.createTimestamp >= :startDate', { startDate })
+      .andWhere('c.createTimestamp <= :endDate', { endDate })
+      .innerJoin('c.project', 'p')
+      .innerJoin('p.district', 'd')
+      .select('p.district_id', 'districtId')
+      .addSelect('d.name', 'districtName')
+      .addSelect('COUNT(c.public_comment_id)', 'publicCommentCount')
+      .groupBy('p.district_id')
+      .addGroupBy('d.name')
+      .orderBy('"publicCommentCount"', 'DESC')
+      .getRawMany();
+  }
+
+  /**
+   * Retrieves the total number of public comments grouped by response code (category).
+   * Used by analytics dashboard module.
+   *
+   * @param startDate - The start of the date range (inclusive, YYYY-MM-DD)
+   * @param endDate - The end of the date range (inclusive, YYYY-MM-DD)
+   * @returns Promise resolving to an array of PublicCommentCountByCategoryResponse
+   */
+  async getCommentCountByResponseCode(
+    startDate: string,
+    endDate: string
+  ): Promise<PublicCommentCountByCategoryResponse[]> {
+    return await this.repository
+      .createQueryBuilder()
+      .where('create_timestamp >= :startDate', { startDate })
+      .andWhere('create_timestamp <= :endDate', { endDate })
+      .select('response_code', 'responseCode')
+      .addSelect('COUNT(public_comment_id)', 'publicCommentCount')
+      .groupBy('response_code')
```

Comment:
The response code could be null

Reply:
Yes, null response code is important - that should be displayed in the front-end, likely as "Not Categorized". See FOM summary page in the app for the wording it uses.